src/main/java/org/spongepowered/server/SpongeVanilla.java

```diff
@@ -54,66 +45,66 @@
 import org.spongepowered.common.SpongeImpl;
 import org.spongepowered.common.SpongeInternalListeners;
 import org.spongepowered.common.entity.ai.SpongeEntityAICommonSuperclass;
+import org.spongepowered.common.inject.SpongeGuice;
 import org.spongepowered.common.interfaces.IMixinServerCommandManager;
 import org.spongepowered.common.network.message.SpongeMessageHandler;
 import org.spongepowered.common.registry.RegistryHelper;
+import org.spongepowered.common.registry.SpongeGameRegistry;
 import org.spongepowered.common.service.permission.SpongeContextCalculator;
 import org.spongepowered.common.service.permission.SpongePermissionService;
 import org.spongepowered.common.service.sql.SqlServiceImpl;
 import org.spongepowered.common.util.SpongeHooks;
 import org.spongepowered.common.world.storage.SpongePlayerDataHandler;
-import org.spongepowered.server.guice.VanillaGuiceModule;
-import org.spongepowered.server.plugin.MetaPluginContainer;
+import org.spongepowered.server.inject.SpongeVanillaModule;
 import org.spongepowered.server.plugin.VanillaPluginManager;
 
 import java.io.IOException;
-import java.util.Optional;
 
-public final class SpongeVanilla extends MetaPluginContainer {
+public final class SpongeVanilla {
 
     public static final SpongeVanilla INSTANCE = new SpongeVanilla();
     public static final MinecraftServer SERVER = SpongeVanillaLauncher.getServer();
 
-    private final SpongeGame game;
+    @Inject private SpongeGame game;
+    @Inject private Logger logger;
+    @Inject private SpongeGameRegistry registry;
+    @Inject private SpongeVanillaContainer container;
     private Cause gameCause;
 
     private SpongeVanilla() {
-        super(SpongeVanillaLauncher.getMetadata().get(SpongeImpl.ECOSYSTEM_ID, "SpongeVanilla"),
-                SpongeVanillaLauncher.findSource(SpongeVanilla.class));
-
         // We force-load NetHandlerPlayServer here.
         // Otherwise, VanillaChannelRegistrar causes it to be loaded from
-        // within the Guice injector (see VanillaGuiceModule), thus swallowing
+        // within the Guice injector (see SpongeVanillaModule), thus swallowing
         // any Mixin exception that occurs.
         //
         // See https://github.com/SpongePowered/SpongeVanilla/issues/235 for a more
         // in-depth explanation
         NetHandlerPlayServer.class.getName();
 
-        Guice.createInjector(new VanillaGuiceModule(this, LogManager.getLogger(SpongeImpl.ECOSYSTEM_NAME))).getInstance(SpongeImpl.class);
-
-        this.game = SpongeImpl.getGame();
+        final Stage stage = SpongeGuice.getInjectorStage(/*VanillaLaunch.ENVIRONMENT == VanillaLaunch.Environment.DEVELOPMENT ? Stage.DEVELOPMENT : */Stage.PRODUCTION);
+        SpongeImpl.getLogger().info("Creating injector in stage '{}'", stage);
```

Comment:
Might be useful as `debug` message, I don't think it's really needed to be always visible